PPAK_CVS/Configuration/Extensions/UIApplication+.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  UIApplication+.swift
+//  PPAK_CVS
+//
+//  Created by 김민지 on 2022/11/20.
+//
+
+import UIKit
+
+extension UIApplication {
+
+  class func topViewController(
+    base: UIViewController? = UIApplication.shared.keyWindow?.rootViewController
+  ) -> UIViewController? {
+    if let presented = base?.presentedViewController {
+      return topViewController(base: presented)
+    }
+    return base
+  }
```

Comment:
1. `class`로 재정의가능하게 한 이유가 궁금하네요. 따로 재정의해서 사용해야할 경우가 있을까요?
2. `keyWindow`는 이미 `ios 13.0`버전에서 **`deprecated`** 되었기에 다른 방법으로 구현하는 게 좋을 거 같아요. 예를 들면, 이런 식인거죠!
 ```suggestion
  var topViewController: UIViewController? {
    (connectedScenes.first as? UIWindowScene)?.windows.first?.rootViewController
  }
```
제가 이렇게 구현한 이유는  `ios 13`부터 UIApplication의 keyWindow가 `deprecated`되는 대신 `connectedScenes` 프로퍼티로 `window`에 접근이 가능하도록 바뀌었고, `windows: [UIWindow]`프로퍼티의 첫 번째는 SceneDelegate에 구현되어있는 `window`이므로 `first` 프로퍼티로 window에 접근하여 가져오도록 구현했습니다.

> Reference
>
> 1. [[iOS] 13 이슈 대응 및 변경 사항 공유](https://jinnify.tistory.com/63)
> 2. [Reference to currently active window](https://developer.apple.com/forums/thread/695115)

Reply:
> 1. `class`로 재정의가능하게 한 이유가 궁금하네요. 따로 재정의해서 사용해야할 경우가 있을까요?
> 2. `keyWindow`는 이미 `ios 13.0`버전에서 **`deprecated`** 되었기에 다른 방법으로 구현하는 게 좋을 거 같아요. 예를 들면, 이런 식인거죠!
> 
> 제가 이렇게 구현한 이유는 `ios 13`부터 UIApplication의 keyWindow가 `deprecated`되는 대신 `connectedScenes` 프로퍼티로 `window`에 접근이 가능하도록 바뀌었고, `windows: [UIWindow]`프로퍼티의 첫 번째는 SceneDelegate에 구현되어있는 `window`이므로 `first` 프로퍼티로 window에 접근하여 가져오도록 구현했습니다.
> 
> > Reference
> > 
> > 1. [[iOS] 13 이슈 대응 및 변경 사항 공유](https://jinnify.tistory.com/63)
> > 2. [Reference to currently active window](https://developer.apple.com/forums/thread/695115)

1. `UIAppliation.topViewController()`로 사용하기 위해서는 클래스 함수로 만들어야 했습니다.
2. 오 그렇군요! 덕분에 `connectedScenes`을 알게 되었습니다👍 적어주신 코드와 여러 블로그들 참고하면서 아래와 같이 수정했는데 어떠신가요??
```swift
  // 변수로 선언
  private var topVC: UIViewController? {
    let scenes = UIApplication.shared.connectedScenes
    let windowScene = scenes.first as? UIWindowScene
    let window = windowScene?.windows.first
    return window?.rootViewController
  }
  ...
  // 사용할 때
  guard let topVC = self.topVC else { return }
```

Reply:
최고에요!